plugin/src/create-schema-customization.ts

```diff
@@ -0,0 +1,228 @@
+import { CreateSchemaCustomizationArgs } from "gatsby";
+
+export function createSchemaCustomization(
+  { actions, schema }: CreateSchemaCustomizationArgs,
+  pluginOptions: ShopifyPluginOptions
+) {
+  const includeCollections =
+    pluginOptions.shopifyConnections &&
+    pluginOptions.shopifyConnections.includes("collections");
+
+  const includeOrders =
+    pluginOptions.shopifyConnections &&
+    pluginOptions.shopifyConnections.includes("orders");
+
+  const name = (name: string) => `${pluginOptions.typePrefix}${name}`;
+
+  const productDef = schema.buildObjectType({
+    name: name("ShopifyProduct"),
+    fields: {
+      variants: {
+        type: `[${name("ShopifyProductVariant")}]`,
+        extensions: {
+          link: {
+            from: "id",
+            by: "productId",
+          },
+        },
+      },
+      images: {
+        type: `[${name("ShopifyProductImage")}]`,
+        extensions: {
+          link: {
+            from: "id",
+            by: "productId",
+          },
+        },
+      },
+    },
+    interfaces: ["Node"],
+  });
+
+  const productImageDef = schema.buildObjectType({
+    name: name("ShopifyProductImage"),
+    fields: {
+      product: {
+        type: name("ShopifyProduct!"),
+        extensions: {
+          link: {
+            from: "productId",
+            by: "id",
+          },
+        },
+      },
+    },
+    interfaces: ["Node"],
+  });
+
+  if (pluginOptions.downloadImages && productImageDef.config.fields) {
+    productImageDef.config.fields.localFile = {
+      type: "File",
+      extensions: {
+        link: {},
+      },
+    };
+  }
+
+  if (includeCollections && productDef.config.fields) {
+    productDef.config.fields.collections = {
+      type: `[${name("ShopifyCollection")}]`,
+      extensions: {
+        link: {
+          from: "id",
+          by: "productIds",
+        },
+      },
+    };
+  }
+
+  const typeDefs = [
+    productDef,
+    productImageDef,
+    schema.buildObjectType({
+      name: name("ShopifyProductVariant"),
+      fields: {
+        product: {
+          type: name("ShopifyProduct!"),
+          extensions: {
+            link: {
+              from: "productId",
+              by: "id",
+            },
+          },
+        },
+        metafields: {
+          type: `[${name("ShopifyMetafield")}]`,
+          extensions: {
+            link: {
+              from: "id",
+              by: "productVariantId",
+            },
+          },
+        },
+      },
+      interfaces: ["Node"],
+    }),
+    /**
+     * FIXME: let's change this to e.g. ShopifyProductVariantMetafield,
```

Comment:
Is this something we are planning to fix in the future? Is there an existing issue or story for this?

Reply:
Yeah I'd like to fix it in the future but haven't created a story. The topic did come up here https://github.com/gatsbyjs/gatsby-source-shopify-experimental/pull/82